core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java

```diff
@@ -0,0 +1,297 @@
+            }
+        }
+        // Recreate the snapshot dir.
+        path.trimTo(snapshotLen).slash$();
+        if (ff.mkdirs(path, configuration.getMkDirMode()) != 0) {
+            throw CairoException.instance(ff.errno()).put("Could not create [dir=").put(path).put(']');
+        }
+
+        try (
+                TableListRecordCursorFactory factory = new TableListRecordCursorFactory(configuration.getFilesFacade(), configuration.getRoot())
+        ) {
+            final int tableNameIndex = factory.getMetadata().getColumnIndex(TableListRecordCursorFactory.TABLE_NAME_COLUMN);
+            try (RecordCursor cursor = factory.getCursor(executionContext)) {
+                final Record record = cursor.getRecord();
+                try (MemoryCMARW mem = Vm.getCMARWInstance()) {
+                    // Copy metadata files for all tables.
+                    while (cursor.hasNext()) {
+                        CharSequence tableName = record.getStr(tableNameIndex);
+                        TableReader reader = engine.getReaderForStatement(executionContext, tableName, "snapshot");
+                        snapshotReaders.add(reader);
+
+                        path.trimTo(snapshotLen).concat(configuration.getDbDirectory()).concat(tableName).slash$();
+                        if (ff.mkdirs(path, configuration.getMkDirMode()) != 0) {
+                            throw CairoException.instance(ff.errno()).put("Could not create [dir=").put(path).put(']');
+                        }
+
+                        int rootLen = path.length();
+                        // Copy _meta file.
+                        path.trimTo(rootLen).concat(TableUtils.META_FILE_NAME).$();
+                        mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                        reader.getMetadata().dumpTo(mem);
+                        mem.close(false);
+                        // Copy _txn file.
+                        path.trimTo(rootLen).concat(TableUtils.TXN_FILE_NAME).$();
+                        mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                        reader.getTxFile().dumpTo(mem);
+                        mem.close(false);
+                        // Copy _cv file.
+                        path.trimTo(rootLen).concat(TableUtils.COLUMN_VERSION_FILE_NAME).$();
+                        mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                        reader.getColumnVersionReader().dumpTo(mem);
+                        mem.close(false);
+
+                        LOG.info().$("snapshot copied [table=").$(tableName).$(']').$();
+                    }
+
+                    // Write instance id to the snapshot metadata file.
+                    path.trimTo(snapshotLen).concat(TableUtils.SNAPSHOT_META_FILE_NAME).$();
+                    mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                    mem.putStr(0, configuration.getSnapshotInstanceId());
```

Comment:
This will set truncate position
```suggestion
                    mem.putStr(configuration.getSnapshotInstanceId());
```

Reply:
Addressed in 3e2f0b4cea1601b70e737b15316c1f8e85b8a3b9